lib/puppet/type/exec.rb

```diff
@@ -197,14 +197,15 @@ def value=(*values)
     end
 
     newparam(:user) do
-      desc "The user to run the command as.  Note that if you
-        use this then any error output is not currently captured.  This
-        is because of a bug within Ruby.  If you are using Puppet to
-        create this user, the exec will automatically require the user,
-        as long as it is specified by name.
-
-        Please note that the $HOME environment variable is not automatically set
-        when using this attribute."
+      desc "The user to run the command as. This does not work on Windows.
```

Comment:
Maybe specify *nix only, instead of "does not work" which is a little unhappy sounding?

Then maybe if there's a windows equivalent, we can suggest that here.

Reply:
I thought about that, but the only thing I know for sure is that it doesn't work on Windows. I'm not sure whether it works on *nix, either. (AIX? Network devices?)

Would it be better to match the error message text instead?
```
        if Puppet.features.microsoft_windows?
self.fail _("Unable to execute commands as other users on Windows")
```